go-minitwit/main.go

```diff
@@ -132,31 +118,38 @@ func gravatarURL(email string, size int) string {
 	return fmt.Sprintf("http://www.gravatar.com/avatar/%x?d=identicon&s=%d", hash, size)
 }
 
-func formatMessages(messages []map[string]interface{}) []Message {
-	var formattedMessages []Message
+func formatMessages(messages []MessageUser) []MessageUI {
+	var formattedMessages []MessageUI
+
+	/*
+		if reflect.TypeOf(m.Text).Kind() == reflect.String {
+			filteredMsg.Content = m.Text
+		}
+	*/
+
 	for _, m := range messages {
-		var msg Message
+		var msg MessageUI
 		// Use type assertion for int64, then convert to int
-		if id, ok := m["message_id"].(int64); ok {
-			msg.MessageID = int(id)
+		if reflect.TypeOf(m.MessageID).Kind() == reflect.Int {
+			msg.MessageID = int(m.MessageID)
 		}
-		if authorID, ok := m["author_id"].(int64); ok {
-			msg.AuthorID = int(authorID)
+		if reflect.TypeOf(m.AuthorID).Kind() == reflect.Int {
+			msg.AuthorID = int(m.AuthorID)
 		}
-		if userID, ok := m["user_id"].(int64); ok {
-			msg.User.UserID = int(userID)
+		if reflect.TypeOf(m.UserID).Kind() == reflect.Int {
+			msg.User.UserID = int(m.UserID)
```

Comment:
why are you typecasting to int here? the ORM will not set it as int already since it reflects an integer?